auth/src/main/java/com/firebase/ui/auth/IdpResponse.java

```diff
@@ -33,20 +34,25 @@
  */
 public class IdpResponse implements Parcelable {
     private final User mUser;
+
     private final String mToken;
     private final String mSecret;
 
-    private final int mErrorCode;
+    @Nullable private final Exception mException;
 
-    private IdpResponse(int errorCode) {
-        this(null, null, null, errorCode);
+    private IdpResponse(Exception e) {
```

Comment:
Should this be `@NonNull`?

Reply:
Yep, I've added another constructor so our nullability can be clearer.